### PR TITLE
UnitTests: Add test TestDeleteNodeAffinityKeyFromNode

### PR DESCRIPTION
**Pull request by iamniting**
Add test for deleteNodeAffinityKeyFromNodes in uninstall_reconciler_test.go

Signed-off-by: Nitin Goyal nigoyal@redhat.com

cc: @raghavendra-talur

### CODE REVIEW

File: pkg/controller/storagecluster/uninstall_reconciler_test.go [hunk 1]
```diff
@@ -131,3 +137,78 @@ func TestDeleteStorageClasses(t *testing.T) {
 		}
 	}
 }
+
+func TestDeleteNodeAffinityKeyFromNodes(t *testing.T) {
```
Comment (umangachapagain) [inline pkg/controller/storagecluster/uninstall_reconciler_test.go:141]: use Table Driven approach and do processing steps  in separate functions outside the main test block for better readability.

File: pkg/controller/storagecluster/uninstall_reconciler_test.go [hunk 2]
```diff
@@ -131,3 +137,78 @@ func TestDeleteStorageClasses(t *testing.T) {
...
+		// delete NodeAffinityKey
+		err = reconciler.deleteNodeAffinityKeyFromNodes(sc, reconciler.reqLogger)
+		assert.NoError(t, err)
+
+		// verify deleteNodeAffinityKeyFromNodes does not delete user defined key
+		assertGetStorageClusterEligibleNodes(t, reconciler, sc)
+	}
+}
+
+func assertGetStorageClusterEligibleNodes(
```
Comment (umangachapagain) [inline pkg/controller/storagecluster/uninstall_reconciler_test.go:207]:
```suggestion
func getStorageClusterEligibleNodes(
```

Comment (umangachapagain) [reply pkg/controller/storagecluster/uninstall_reconciler_test.go:207]: If you need to assert values of  "node", do it after function call.

**Review by umangachapagain — CHANGES_REQUESTED** [review]

File: pkg/controller/storagecluster/uninstall_reconciler_test.go (hunk 1 above)
Comment (iamniting) [reply pkg/controller/storagecluster/uninstall_reconciler_test.go:164]: Done

File: pkg/controller/storagecluster/uninstall_reconciler_test.go (hunk 2 above)
Comment (iamniting) [reply pkg/controller/storagecluster/uninstall_reconciler_test.go]: Deleted the func as I have mainly created the func for the assert and we already have a func for getting nodes

File: pkg/controller/storagecluster/uninstall_reconciler_test.go [hunk 3]
```diff
@@ -154,3 +160,89 @@ func assertTestDeleteStorageClasses(t *testing.T, reconciler ReconcileStorageClu
...
+func TestDeleteNodeAffinityKeyFromNodes(t *testing.T) {
+
+	testList := []struct {
+		createUserDefinedNodeAffinityKey bool
+	}{
+		// case1: verify deleteNodeAffinityKeyFromNodes deletes default NodeAffinityKey only
+		{false},
+		// case2: verify deleteNodeAffinityKeyFromNodes does not deletes user defined NodeAffinityKey
+		{true},
+	}
```
Comment (jarrpa) [inline pkg/controller/storagecluster/uninstall_reconciler_test.go:173]: As in https://github.com/openshift/ocs-operator/pull/794#discussion_r505915347 , make this a bit more readable.

**Review by jarrpa — CHANGES_REQUESTED** [review]

**Comment by jarrpa** [comment]
Also, remove the StorageClasses testing commit, that's not part of this PR.

File: pkg/controller/storagecluster/uninstall_reconciler_test.go (hunk 3 above)
Comment (iamniting) [reply pkg/controller/storagecluster/uninstall_reconciler_test.go]: Done

**Comment by iamniting** [comment]
> Please remove the StorageClasses testing commit, that's not part of this PR. Also, you have the wrong subject and message on your commit, it should be for "UnitTests: Add test TestDeleteNodeAffinityKeyFromNode".

Actually, I was doing like this, So I don't have to rebase it later. As changes are in one file and there will be merge conflicts later. Anyway, I have removed those changes I will rebase them later itself.
I have changed the commit-msg.

**Comment by iamniting** [comment]
Closing it as it is part of https://github.com/openshift/ocs-operator/pull/807

**Pull request closed without merging**